### PR TITLE
Add multi-gpu support to time

### LINKED ISSUE #3484: cupyx.time.repeat does not work for multi-gpu benchmark
Currently, events are created and synchronized only for the current device in `cupyx.time.repeat`.

### PR DESCRIPTION
**Pull request by emcastillo**
Closes #3484 

Check with

```python
import cupy
import cupyx
import time


size = 10 ** 8


def single_device():
    with cupy.cuda.Device(0):
        arr_0 = cupy.ones(size)
        arr_1 = cupy.ones(size)
        arr_0 * arr_1


def multi_device():
    streams = []
    with cupy.cuda.Device(0):
        arr_0_0 = cupy.ones(size // 2)
        arr_0_1 = cupy.ones(size // 2)

    with cupy.cuda.Device(1):
        arr_1_0 = cupy.ones(size // 2)
        arr_1_1 = cupy.ones(size // 2)

    with cupy.cuda.Device(0):
        arr_0_0 * arr_0_1

    with cupy.cuda.Device(1):
        arr_1_0 * arr_1_1


print(cupyx.time.repeat(single_device, n_repeat=10))
print(cupyx.time.repeat(multi_device, n_repeat=10, devices=[0,1]))
```

### CODE REVIEW

File: cupyx/time.py [hunk 1]
```diff
@@ -64,36 +72,66 @@ def repeat(
...
-    ev1.record()
-    ev1.synchronize()
+    for event, device in zip(events_1, devices):
+        with cupy.cuda.Device(device):
+            event.record()
+        event.synchronize()
 
     cpu_times = []
-    gpu_times = []
+    gpu_times = [[] for i in range(len(events_1))]
```
Comment (asi1024) [inline cupyx/time.py:105]:
```suggestion
    gpu_times = [[] for i in events_1]
```

File: cupyx/time.py [hunk 2]
```diff
@@ -64,36 +72,66 @@ def repeat(
...
+            with cupy.cuda.Device(device):
+                event.record()
+        for event, device in zip(events_2, devices):
+            with cupy.cuda.Device(device):
+                event.synchronize()
+        i = 0
+        for ev1, ev2 in zip(events_1, events_2):
+            gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
+            gpu_times[i].append(gpu_time)
+            i += 1
```
Comment (asi1024) [inline cupyx/time.py:130]:
```suggestion
        for i, (ev1, ev2) in enumerate(zip(events_1, events_2)):
            gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
            gpu_times[i].append(gpu_time)
```

File: cupyx/time.py [hunk 3]
```diff
@@ -8,20 +8,21 @@
 
 
 class _PerfCaseResult(object):
-    def __init__(self, name, ts):
+    def __init__(self, name, ts, devices):
         assert ts.ndim == 2
-        assert ts.shape[0] == 2
+        assert ts.shape[0] >= 2
```
Comment (asi1024) [inline cupyx/time.py:13]:
```suggestion
        assert ts.shape[0] == len(devices) + 1
```

File: cupyx/time.py (hunk 2 above)
Comment (emcastillo) [reply cupyx/time.py:130]: I didn't know I could expand the tuple like this 😬
awesome!

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**